src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentStateCopy = { ...state };
+
+  for (const action of actions) {
+    const stateCopy = { ...currentStateCopy };
+
+    if (action.type === 'addProperties') {
+      for (const [key, value] of Object.entries(action.extraData)) {
+        stateCopy[key] = value;
+      }
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete stateCopy[key];
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const key in stateCopy) {
+        delete stateCopy[key];
+      }
+    }
```

Comment:
[CODE STYLE]: Use switch statement if you have limited amount of conditions. It will make the code cleaner and more maintainable. Replace the if statements with a switch statement that handles the different types of actions.